### ISSUE TITLE
`redundant_self` false positive in 0.63.0-rc.2 (optional extension)

### ISSUE THREAD

**Issue by shagedorn** [realm/SwiftLint#6390]
### New Issue Checklist

- [x] I've Updated SwiftLint to the latest version.
- [x] I've searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues).

### Bug Description

When writing an extension to an optional type and you first unwrap `self`, the compiler requires you to explicitly use `self` going forward, because implicit `self` is the optional type.

```
extension String? {
    func foo() -> String {
        guard let self else { return "" }
        return self.uppercased()
    }
}
```

This code compiles, but throws a `redundant_self` linter warning.

When removing `self`, the compiler generates a `Cannot find 'uppercased' in scope` warning.

### Environment

SwiftLint version: 0.63.0-rc.2
Xcode version: 26.2
Installation method used: release ZIP file download
Configuration file: irrelevant, except that `redundant_self` is enabled

**Comment by shagedorn** [comment]
Thanks for the quick fix ❤️
